src/App.tsx

```diff
@@ -1,23 +1,32 @@
-import React, { useState } from 'react';
-// import AddInvoiceElement from './Components/AddInvoiceElement';
+import { getAuth, onAuthStateChanged } from 'firebase/auth';
+import { useEffect, useState } from 'react';
 import HomePage from './Components/HomePage';
-import LoginPage from './Components/LoginPage';
-import ModalWindow from './Components/ModalWindow';
-import SignUpPage from './Components/SignUpPage';
-import { cteateUserInFirebaseWithEmailAndPassword, loginToFirebaseWithEmailAndPassword, logoutFromFirebase } from './dataBaseUtils/auth/authUtils';
+import LoginPage from './Components/loginAndSignup/LoginPage';
+import SignUpPage from './Components/loginAndSignup/SignUpPage';
 import "./dataBaseUtils/auth/firebaseInit.ts"
-import { writeIntoDataBase } from './dataBaseUtils/readWrite';
 
 function App() {
-	const [signedIn, setSignedIn] = useState<boolean>(false);
-	const [currentPage, setCurrentPage] = useState<string>("login");
-	// loginToFirebaseWithEmailAndPassword("treex.lights@gmail.com", "123456");
-	return (
-		<div >
-			{currentPage === "login" && (!signedIn ? <LoginPage setSignedIn={setSignedIn} setCurrentPage={setCurrentPage} /> : <HomePage setSignedIn={setSignedIn} />)}
-			{currentPage === "signup" && (!signedIn ? <SignUpPage setSignedIn={setSignedIn} setCurrentPage={setCurrentPage} /> : <HomePage setSignedIn={setSignedIn} />)}
-		</div>
-	);
+  const [signedIn, setSignedIn] = useState<boolean>(false);
+  const [currentPage, setCurrentPage] = useState<string>("login");
+  useEffect(() => {
+    const isUser = sessionStorage.getItem('user');
+    if (isUser) {
+      const auth = getAuth();
+      onAuthStateChanged(auth, (user) => {
+        if (user) {
+          setSignedIn(true);
+        } else {
+          sessionStorage.removeItem("user");
+        }
+      });
+    }
+  }, [])
+  return (
+    <div >
+      {currentPage === "login" && (!signedIn ? <LoginPage setSignedIn={setSignedIn} setCurrentPage={setCurrentPage} /> : <HomePage setSignedIn={setSignedIn} />)}
+      {currentPage === "signup" && (!signedIn ? <SignUpPage setSignedIn={setSignedIn} setCurrentPage={setCurrentPage} /> : <HomePage setSignedIn={setSignedIn} />)}
+    </div>
+  );
```

Comment:
Normally you do things like these with `react-router`. 

If you want to do it yourself it's totally fine. But in this case I'd suggest to add a bit of finesse to this structure.

You could define on the top of this file something like:

``` 
const unauthorizedComponentsMap = {
  // If you made RoutePath enum
  [RoutePath.Login]: LoginPage,
  [RoutePath.SignUp]: SignUpPage,
};
```

and then in the component:
```
const UnauthorizedComponent = unauthorizedComponentsMap[currentPage];

if (!signedIn) return <UnauthorizedComponent setSignedIn={setSignedIn} setCurrentPage={setCurrentPage} />;

return <HomePage setSignedIn={setSignedIn} />
```

It always makes sense to think of how you can reduce the amount of complexity in your code. And conditionals are the main sources of introduced complexity.

Reply:
Thanks for the example. I did not know that CompomentMap way of doing these things. Changed to it. Looks more digestible now.